torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1210,10 +1210,16 @@ void ProcessGroupNCCL::heartbeatMonitor() {
   // we throw exception and make the whole process to be killed.
   const auto exitMsg = c10::str(
       logPrefix(),
-      "NCCL monitor thread timeout. Basically, this could ",
-      "be due to CUDA or NCCL calls being unexpectedly blocking, ",
-      "especially when your program enters a deadlock state in watchdog "
-      "or destructors. If you see this error, please file a bug to PyTorch.");
+      "ProcessGroupNCCL's watchdog got stuck for ",
+      heartbeatTimeoutInSec_,
+      "seconds without making progress in monitoring enqueued collectives. ",
+      "This typically indicates a nccl/cuda API hang blocking the watchdog, ",
+      "and could be triggered by another thread holding the GIL inside a ",
+      "cuda api, or other deadlock-prone behaviors.",
+      "If you suspect the watchdog is not actually stuck and a longer timeout would help, ",
+      "you can either increase the timeout(TORCH_NCCL_HEARTBEAT_TIMEOUT_SEC) to a larger value "
+      "or disable the heartbeat monitor(TORCH_NCCL_ENABLE_MONITORING=0)"
```

Comment:
missing a space before (

in 2 places (after timeout, then again after monitor)

Reply:
Actually I thought about it, I thought no space is actually preferred.. No strong opinion here.